### ISSUE TITLE
Issues with rejoining a conference on Firefox

### ISSUE THREAD

**Issue by seanmheff** [jitsi/jitsi-meet#367]
I have a conference with two participants (A and B). Each participant is running Firefox 41.0.1.

If the following sequence of events happen - B's video will freeze on A's window.

A joins. B joins.

B refreshes the page.

A mutes video.

**Comment by seanmheff** [comment]
This problem does not occur with Chrome

**Comment by gpolitis** [comment]
Hi @seanmheff could you please try with https://beta.meet.jit.si/? We recently fixed a very similar issue (if not the same) but I don't think it's deployed in https://meet.jit.si/ yet.

**Comment by seanmheff** [comment]
Ok will do. How often is the code on beta.meet.jit.si updated?

**Comment by gpolitis** [comment]
It runs the latest nightly as long as it passes the tests.

**Comment by damencho** [comment]
It runs the latest versions, no matter whether the tests has passed or not :)

**Comment by gpolitis** [comment]
I stand corrected, thanks @damencho :-)

**Comment by seanmheff** [comment]
I tried to test this out on the beta server. Firefox is not working at all anymore. I see a black screen.

It looks like it is not asking for video in the getUserMedia request.

Chrome still worksl

<img width="682" alt="screen shot 2015-10-07 at 10 16 33 am" src="https://cloud.githubusercontent.com/assets/2361700/10333585/d1038cd2-6cdc-11e5-971f-0c7b0ae4ee25.png">

**Comment by gpolitis** [comment]
Thanks for trying. Are you trying to join without video?

**Comment by seanmheff** [comment]
OK, I can now have a conference with Firefox. I just retried it again and it worked. I did restart Firefox since testing again. Maybe something was amiss there. I am using the built-in mic and camera on a Macbook.

I cannot retest my original problem as I can't make it that far. Here is the issue I now run into:

A joins. B joins.

B refreshes the page.

A no longer see's B's video. B shows as a participant but they just have a black tile. Tested this a few times and I always got the same result.

**Comment by gpolitis** [comment]
Thanks for trying again. Can you try this using the same FF window with A and B on different tabs? Are you on a wired or on a wireless connection? How much have you waited for B's video to show up again? Do you get any error messages in the JS console?

**Comment by seanmheff** [comment]
Hey there. I tried it again using the same FF window and different tabs. I got video to play after refreshing - after about 100 seconds of a black tile.

I have a pretty poor bandwidth at my location so I am running the Jitsi stack on a Ubuntu virtual machine on my Mac.

I did some investigation, it looks like when B refreshes his video, the video does not become active for A until about 100 seconds. I added some extra timestamps to the logging code to verify this. Here are two screenshots of this happening:

![screen shot 2015-10-08 at 11 17 34 am](https://cloud.githubusercontent.com/assets/2361700/10365902/8960b606-6dbe-11e5-9f70-e52c4cd119ab.png)
![screen shot 2015-10-08 at 11 14 13 am](https://cloud.githubusercontent.com/assets/2361700/10365903/896720fe-6dbe-11e5-86b6-49157f451c23.png)


On Chrome, B's video becomes active almost instantly:

![screen shot 2015-10-08 at 11 27 25 am](https://cloud.githubusercontent.com/assets/2361700/10365942/e443e48a-6dbe-11e5-8b85-0e9b80c56498.png)
